chips/sam4l/src/chip.rs

```diff
@@ -152,4 +153,16 @@ impl Chip for Sam4l {
     fn systick(&self) -> &cortexm4::systick::SysTick {
         self.systick
     }
+
+    fn prepare_for_sleep(&self) {
+        if pm::deep_sleep_ready() {
+            unsafe {
+                cortexm4::scb::set_sleepdeep();
+            }
+        } else {
+            unsafe {
+                cortexm4::scb::unset_sleepdeep();
```

Comment:
Is this the line of code that somehow wakes us up? Otherwise, why are you unsetting deep sleep here instead of just doing nothing?